bids2openminds/main.py

```diff
@@ -2,18 +2,70 @@
 import os
 import pathlib
 from warnings import warn
-
+from bids import BIDSLayout
 import pandas as pd
 from nameparser import HumanName
 
 import openminds.latest.core as omcore
 import openminds.latest.controlled_terms as controlled_terms
 from openminds import IRI
 
-from .utility import table_filter, pd_table_value, file_hash, file_storage_size
+from .utility import table_filter, pd_table_value, file_hash, file_storage_size, read_json
 from .mapping import bids2openminds_instance
 
 
+def read_bids_directory(input_path):
+    """
+    Reads the content of the dataset_description.json file and determines whether the dataset is raw, derivative, or contains derivatives. Then, it calls the appropriate BIDSLayout function.
+    Returns three Python objects.
+
+    Parameters:
+    - input_path: The path to the BIDS folder.
+
+    Returns:
+    - bids_layout: An instance of the Layout class from pybids for this dataset.
+    - layout_df: A pandas DataFrame containing all the files and corresponding information.
+    - dataset_description: A Python dictionary containing the content of the dataset_description JSON file.
+    """
+
+    if not os.path.isdir(input_path):
+        raise NotADirectoryError(
+            f"The input path is not valid. You specified '{input_path}', which is not a directory.")
+
+    # if not(BIDSValidator().is_bids(input_path)):
```

Comment:
delete commented lines or specify what the missing TODO / FIXME is for those lines